File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ProgressBar/Functionality/IncrementPage.vb
Namespace CompanionKit.Controls.ProgressBar.Functionality

	Public Class IncrementPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

		End Sub

      ''' <summary>
        ''' Handles the Click event of the button that increments value of the ProgressBar.
        ''' </summary>
        Private Sub mobjIncrementButton_Click(sender As Object, e As EventArgs) Handles mobjIncrementButton.Click
            If Me.mobjDemoProgressBar.Value = Me.mobjDemoProgressBar.Maximum Then
                Me.mobjDemoProgressBar.Value = 0
            End If
            ' Compute step of increasing ProgressBar value.
            Dim mintStep As Integer = Decimal.ToInt16(Me.mobjStepProgressBarNumericUpDown.Value)
            If Me.mobjDemoProgressBar.Value + mintStep > Me.mobjDemoProgressBar.Maximum Then
                mintStep = Me.mobjDemoProgressBar.Maximum - Me.mobjDemoProgressBar.Value
            End If
            ' Increment value of the demonstrated ProgressBar.
            Me.mobjDemoProgressBar.Increment(mintStep)
        End Sub
    End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ProgressBar/Functionality/IncrementPage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.ProgressBar.Functionality
{
    public partial class IncrementPage : UserControl
    {
        public IncrementPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles the Click event of the button that increments value of the ProgressBar.
        /// </summary>
        private void mobjIncrementButton_Click(object sender, EventArgs e)
        {
            if (this.mobjDemoProgressBar.Value == this.mobjDemoProgressBar.Maximum)
            {
                this.mobjDemoProgressBar.Value = 0;
            }
            // Compute step of increasing ProgressBar value.
            int mintStep = decimal.ToInt16(this.mobjStepProgressBarNumericUpDown.Value);
            if (this.mobjDemoProgressBar.Value + mintStep > this.mobjDemoProgressBar.Maximum)
            {
                mintStep = this.mobjDemoProgressBar.Maximum - this.mobjDemoProgressBar.Value;
            }
            // Increment value of the demonstrated ProgressBar.
            this.mobjDemoProgressBar.Increment(mintStep);
        }

    }
}